REGRESSION(298125@main): [WPE][Qt6] fix build

fujii opened the pull request:
#### 81ec41b30d6f24ecf61b0d2f619bf9b17b0dd14d
<pre>
REGRESSION(298125@main): [WPE][Qt6] fix build
<a href="https://bugs.webkit.org/show_bug.cgi?id=297395">https://bugs.webkit.org/show_bug.cgi?id=297395</a>

Reviewed by Michael Catanzaro and Carlos Garcia Campos.

After &lt;<a href="https://commits.webkit.org/298125@main">https://commits.webkit.org/298125@main</a>&gt; changed get_drm_device
and removed get_drm_render_node of wpe_display class,
qt6/WPEDisplayQtQuick.cpp couldn&apos;t compile.

* Source/WebKit/UIProcess/API/wpe/qt6/WPEDisplayQtQuick.cpp:
(wpeDisplayQtQuickConnect):
(wpeDisplayQtQuickGetDRMDevice):
(wpe_display_qtquick_class_init):
(wpeDisplayQtQuickGetDRMRenderNode): Deleted.

Canonical link: <a href="https://commits.webkit.org/298741@main">https://commits.webkit.org/298741@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ed13ba861e47b67ff118e890ed0ab73362690b3b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/116536 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/131/builds/36197 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/138/builds/26750 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/122595 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/67100 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/36891 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/123/builds/44785 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/122595 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/67100 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/119485 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/29422 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/138/builds/26750 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/122595 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/28487 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/138/builds/26750 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/66274 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/98799 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/138/builds/26750 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/125742 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/128/builds/43430 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/123/builds/44785 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/125742 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/121/builds/43795 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/138/builds/26750 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/125742 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/138/builds/26750 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/39384 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18604 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/127/builds/43318 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/48909 "Built successfully") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/42784 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/46124 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/44489 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Source/WebKit/WPEPlatform/wpe/WPEDRMDevicePrivate.h:
```diff
@@ -28,4 +28,4 @@
 #include "WPEDRMDevice.h"
 
-GRefPtr<WPEDRMDevice> wpeDRMDeviceCreateForDevice(const char*);
+WPE_API GRefPtr<WPEDRMDevice> wpeDRMDeviceCreateForDevice(const char*);
```
We don't want to export this symbol.

carlosgcampos commented on Source/WebKit/UIProcess/API/wpe/qt6/WPEDisplayQtQuick.cpp:
```diff
@@ -75,21 +76,24 @@ static gboolean wpeDisplayQtQuickConnect(WPEDisplay* display, GError** error)
         return FALSE;
     }
 
+    priv->drmDevice = wpeDRMDeviceCreateForDevice(drmRenderNode.isNull() ? drmRenderNode.data() : drmDevice.data());
```
```suggestion
    priv->drmDevice = adoptGRef(wpe_drm_device_new(drmDevice.data(), drmRenderNode.data()));
```

fujii replied:
Great! Thank you.

carlosgcampos commented on Source/WebKit/UIProcess/API/wpe/qt6/WPEDisplayQtQuick.cpp:
```diff
@@ -47,8 +47,7 @@
     EGLDisplay eglDisplay;
-    CString drmDevice;
-    CString drmRenderNode;
+    GRefPtr<WPEDRMDevice> drmDevice;
```
I had forgotten, to use GRefPtr with WPEDRMDevice you need to include <wpe/GRefPtrWPE.h>

fujii replied:
Fixed.

carlosgcampos reviewed (approved):
Looks good now, but do not forget to add the missing include before landing. Thanks!

fujii committed:
REGRESSION(298125@main): [WPE][Qt6] fix build
https://bugs.webkit.org/show_bug.cgi?id=297395

Reviewed by Michael Catanzaro and Carlos Garcia Campos.

After <https://commits.webkit.org/298125@main> changed get_drm_device
and removed get_drm_render_node of wpe_display class,
qt6/WPEDisplayQtQuick.cpp couldn't compile.

* Source/WebKit/UIProcess/API/wpe/qt6/WPEDisplayQtQuick.cpp:
(wpeDisplayQtQuickConnect):
(wpeDisplayQtQuickGetDRMDevice):
(wpe_display_qtquick_class_init):
(wpeDisplayQtQuickGetDRMRenderNode): Deleted.

Canonical link: https://commits.webkit.org/298741@main
